cpp/src/lance/arrow/dataset.cc

```diff
@@ -316,6 +317,41 @@ ::arrow::Result<std::shared_ptr<UpdaterBuilder>> LanceDataset::NewUpdate(
                                           std::move(new_field));
 }
 
+::arrow::Result<std::shared_ptr<LanceDataset>> LanceDataset::AddColumn(
```

Comment:
again the problem here is if the compute expression contains aggregates

Reply:
This can be checked via `bool Expression::IsScalarExpression() const`.  

I can throw a invalid status from AddColumn